common/src/main/java/apoc/export/util/NodesAndRelsSubGraph.java

```diff
@@ -63,6 +63,16 @@ public Iterable<IndexDefinition> getIndexes() {
         return indexes;
     }
 
+    @Override
+    public Iterable<ConstraintDefinition> getConstraints() {
+        Schema schema = tx.schema();
+        ArrayList<ConstraintDefinition> constraints = new ArrayList<>(labels.size() * 2);
+        for (String label : labels) {
+            Iterables.addAll(constraints, schema.getConstraints(Label.label(label)));
+        }
+        return constraints;
+    }
```

Comment:
Moreover, I'm pretty sure both `apoc.export.cypher.data` and `apoc.export.cypher.graph` (which use `NodesAndRelsSubGraph.java`) don't work correctly with schemas, 
as only `labels` are looped here, not `types`.

It could be solved in this pr, 
but maybe it's better to create a separate card for the other procedures and just leave the `all` here, 
wdyt?

Reply:
Added the card about data and graph: https://trello.com/c/d0LvEgnK/2193-check-the-correct-behaviour-with-indexes-constraint-using-the-apocexportcypherdata-graph-procedures

Reply:
I agree, we have to add sorting of constraints to avoid flakyness